dual_net.py

```diff
@@ -41,14 +41,19 @@
 # How many positions can fit on a graphics card. 256 for 9s, 16 or 32 for 19s.
 TRAIN_BATCH_SIZE = 256
 
+# The shuffle buffer size determines how far an example could end up from
+# where it started; this and the interleave parameters in preprocessing can give
+# us an approximation of a uniform sampling.  The default of 4M is used in
+# training, but smaller numbers can be used for aggregation or validation.
+SHUFFLE_BUFFER_SIZE = int(4*1e6)
```

Comment:
definitely not a fan of this, but am okay with it pending a better solution for hyperparams.

Reply:
Ok, i put it back in preprocessing and had the func bodies check for =None, which lets us monkeypatch in a new default (damn python's sticky defaults)